application/admin/views.py

```diff
@@ -247,11 +247,23 @@ def merge_data_sources():
 
 @admin_blueprint.route("/data-sources/merge", methods=["POST"])
 @login_required
-@user_can(MANAGE_USERS)  # TODO: update to MANAGE_DATA_SOURCES
+@user_can(MANAGE_DATA_SOURCES)
 def merge_data_sources_post():
 
     data_source_ids = request.form.getlist("ids")
-    data_source_to_keep = DataSource.query.get(request.form.get("keep"))
+
+    data_source_to_keep_id = request.form.get("keep")
```

Comment:
It's a shame not to consistently use our forms framework, because it handles things like this for us (supposedly), including the validation and converting to the right data types. It might be a bit painful to learn but I think it's for the best. There are plenty of examples across the code base of how to do it.

Reply:
Can you help with this? I got a bit stuck trying to understand the framework (particularly for hidden fields).